MekHQ/src/mekhq/campaign/parts/Refit.java

```diff
@@ -1433,20 +1432,37 @@ else if (part instanceof MissingPart) {
                 suit.setUnit(oldUnit);
             }
         }
+        int expectedHeatSinkParts = 0;
+        if ((newEntity instanceof Aero) && ((newEntity.getEntityType() &
+                (Entity.ETYPE_CONV_FIGHTER | Entity.ETYPE_SMALL_CRAFT | Entity.ETYPE_JUMPSHIP)) == 0)) {
```

Comment:
The determination  of "is aero but not conventional fighter/smallcraft/jumpship" is used multiple times in this file, here and on line 2480. And probably in other places. Let's place it into a separate function for ease of reuse.

Reply:
I wasn't a fan of this either. I searched around to see if there was an operator similar to instanceof that didn't match on sub-classes. There doesn't appear to be one, however I ran across a post that suggested doing the equivalent of "newEntity.getClass() == Aero.class", which seems like it would work. In the end, I just went with the method previously used (I think the one you mentioned at line 2480).

"newEntity.getClass() == Aero.class" is technically more precise in that it would exclude TeleMissile and FighterSquadron as well (not sure how though are used).

Would you prefer I move the original to a function or use the "newEntity.getClass() == Aero.class" method?

Reply:
Let's give that getClass() == Aero.class thing a try. Just make sure to beat up on that code with other non-ASF unit types as well.

Reply:
Tested and works well.